src/Microsoft.Diagnostics.Monitoring.RestServer/DiagnosticServices.cs

```diff
@@ -50,6 +50,12 @@ public async Task<IEnumerable<IProcessInfo>> GetProcessesAsync(CancellationToken
                 IList<Task<ProcessInfo>> processInfoTasks = new List<Task<ProcessInfo>>();
                 foreach (IEndpointInfo endpointInfo in await _endpointInfoSource.GetEndpointInfoAsync(token))
                 {
+                    // CONSIDER: Can this processing be pushed into the IEndpointInfoSource implementation and cached
```

Comment:
Outside the reverse scenario, I think we could only reliably cache entries for runtime id based queries.

Reply:
I have a proposal for generating instance cookies for .NET Core 3.1 that is stable for the lifetime of each process based on the diagnostic pipe name components. Then that or the runtime instance cookie can be used for identifying cached items. But we don't have that in place yet.